stripe-patrons-data/src/main/scala/com/gu/patrons/services/StripeSubscriptionsProcessor.scala

```diff
@@ -16,18 +16,19 @@ class StripeSubscriptionsProcessor(
 ) {
 
   def processSubscriptions(pageSize: Int) =
-    processFutureResponse(stripeService.getSubscriptions(pageSize))
+    processFutureResponse(pageSize, None)
 
-  final def processFutureResponse(eventualResponse: Future[StripeSubscriptionsResponse]): Future[Unit] = {
-    eventualResponse.flatMap { response =>
-      processSubs(response.data).map(_ =>
```

Comment:
turns out the issue this should have been flatMap, not map, and so the Futures were not combined